packages/components/src/SideDrawer/SideDrawer.tsx

```diff
@@ -155,6 +182,49 @@ export function SideDrawer({
     }
   }
 }
+const transition = {
+  duration: tokens["timing-base"] / 1000,
+  ease: "easeInOut",
+} as const;
+const BUTTON_WIDTH = 40;
+
+function BackButtonContainer(props: HTMLAttributes<HTMLDivElement>) {
+  const { components } = useSideDrawerContext();
+  const [scope, animate] = useAnimate<HTMLDivElement>();
+
+  useEffect(() => {
+    if (components.backButton) {
+      animate(
```

Comment:
Could we have built this with CSS Animations instead and applying classes? This feels like a lot of work for what might have been a ~5 line keyframe animation.

Reply:
actually, since i'm animating between known values, probs should be actually. lemme try that